### PR TITLE
feat: add quality tests

### PR DESCRIPTION
**Pull request by JuanDavidBuitrago**
Description:
This PR adds QA to maintain the repo

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        make requirements
+    - name: Run the QA suite
+      run: |
+        make quality
+    # - name: Unit tests
+    #   run: |
+    #     make test
```
Comment (mariajgrimaldi) [inline .github/workflows/ci.yml]: why is this commented?

Comment (JuanDavidBuitrago) [reply .github/workflows/ci.yml]: I uncommented it. It works!!

File: tutorcodejail/plugin.py [hunk 2]
```diff
@@ -18,29 +19,28 @@
...
-        "codejail_apparmor": f"docker.io/ednxops/codejail_apparmor:{__version__}"
+        "codejail_apparmor": f"docker.io/ednxops/codejail_apparmor:{__version__}",
     },
-    "init": ["codejail_apparmor"]
+    "init": ["codejail_apparmor"],
 }
 
 
 def patches():
+    """Logic for retrueve all the patches of tutorcodejail."""
```
Comment (johanseto) [inline tutorcodejail/plugin.py]: Retrieve hehe, and I think you can quit the logic and only keep Retrieve.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: Unit tests, QA
+
+on:
+  push:
+    branches: [ main ]
```
Comment (johanseto) [inline .github/workflows/ci.yml]: Could you change branches to  **. I think that helps for development purposes.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+    branches: [ main ]
```
Comment (johanseto) [inline .github/workflows/ci.yml]: Here too the '**' to target all branches' PR. Or could be considered 🤔.

 Also maybe you can use common YAML syntax for lists. Or do you prefer the abbreviated way? https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html

File: Makefile [hunk 5]
```diff
@@ -50,7 +50,7 @@ requirements: ## install development environment requirements
 	pip install -e .
 
 test: ## run unitary tests and meassure coverage
-	coverage run -m pytest
```
Comment (mariajgrimaldi) [inline Makefile:53]: why is this needed?

File: tutorcodejail/tests/test_patches.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+"""Tests to verify number of patches"""
+
+
+from tutorcodejail.plugin import patches
+
+
+def test_patches():
+    all_patches = patches()
+
+    assert len(all_patches) == 5
```
Comment (mariajgrimaldi) [inline tutorcodejail/tests/test_patches.py]: Please, fix this test:
1. The module docstring must describe what the module does not the test implemented in it
2. Use the unittest Python library
3. Add docstrings for each test

File: .github/workflows/ci.yml (hunk 4 above)
Comment (johanv26) [reply .github/workflows/ci.yml]: What do you think of the YAML syntax?
branches:
  - main

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+name: Unit tests, QA
+
+on:
+  push:
+    branches: [ '**' ]
```
Comment (johanseto) [inline .github/workflows/ci.yml]: Also maybe you can use common YAML syntax for lists. Or do you prefer the abbreviated way? https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html
What do you think of the YAML syntax? 
@mariajgrimaldi @JuanDavidBuitrago 
e.g.
```
branches:
  - main
  - '**'
```

Comment (JuanDavidBuitrago) [reply .github/workflows/ci.yml]: You're right @johanv26 !! Is better use an array list.

File: tutorcodejail/tests/test_patches.py (hunk 6 above)
Comment (JuanDavidBuitrago) [reply tutorcodejail/tests/test_patches.py]: I already fixed it

File: tutorcodejail/tests/test_plugin.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    Test case for tutor codejail plugin.
+    """
+
+    def test_patches(self):
+        """
+        Test total patches for codejail plugin.
+
+        Expected results:
+            - Total patches expected from patches dir
```
Comment (mariajgrimaldi) [inline tutorcodejail/tests/test_plugin.py:17]: instead of checking the number, you could which one returns

File: tutorcodejail/tests/test_plugin.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        """
+
```
Comment (mariajgrimaldi) [inline tutorcodejail/tests/test_plugin.py:26]: remove this line please

**Review by mariajgrimaldi — APPROVED** [review]

File: tutorcodejail/tests/test_plugin.py (hunk 8 above)
Comment (JuanDavidBuitrago) [reply tutorcodejail/tests/test_plugin.py:17]: Done

File: tutorcodejail/tests/test_plugin.py (hunk 9 above)
Comment (JuanDavidBuitrago) [reply tutorcodejail/tests/test_plugin.py:26]: Done!!
